test/test_nn.py

```diff
@@ -13399,6 +13399,264 @@ def test_linear_state(self):
         with self.assertRaisesRegex(RuntimeError, 'shape of an uninitialized'):
             module.load_state_dict(lazy_module.state_dict())
 
+    @suppress_warnings
+    def test_lazy_conv1d(self):
+        module = nn.LazyConv1d(32, 2)
+        self.assertIsInstance(module.weight, UninitializedParameter)
+        input = torch.ones(192, 16, 50)
+        module(input)
+        self.assertIsInstance(module, nn.Conv1d)
+        self.assertNotIsInstance(module, nn.LazyConv1d)
+        self.assertEqual(module.weight.shape, (32, 16, 2))
+        y = module(input)
+        self.assertTrue(torch.equal(torch.nn.functional.conv1d(input, module.weight, module.bias), y))
+
+    @suppress_warnings
+    def test_lazy_conv1d_pickle(self):
+        module = nn.LazyConv1d(32, 2)
+        self.assertIsInstance(module.weight, UninitializedParameter)
+        module = pickle.loads(pickle.dumps(module))
+        self.assertIsInstance(module, nn.LazyConv1d)
+        self.assertIsInstance(module.weight, UninitializedParameter)
+        input = torch.ones(192, 16, 50)
+        module(input)  # fully materialized
+        new_module = pickle.loads(pickle.dumps(module))
+        self.assertIsInstance(new_module, nn.Conv1d)
+        self.assertNotIsInstance(new_module, nn.LazyConv1d)
+        self.assertEqual(new_module.weight.shape, (32, 16, 2))
+        self.assertNotIsInstance(new_module.weight, UninitializedParameter)
+
+    @suppress_warnings
+    def test_lazy_conv1d_state(self):
+        module = nn.Conv1d(16, 32, 2)
+        lazy_module = nn.LazyConv1d(32, 2)
+        lazy_module.load_state_dict(module.state_dict())
+        # Parameters have been initialized but the module won't become a full
+        # Conv1d one until the first iteration. This is due to
+        # limitations on the state_dict loading logic
+        self.assertFalse(lazy_module.has_uninitialized_params())
+        self.assertEqual(lazy_module.weight.shape, (32, 16, 2))
+
+        module = nn.Conv1d(16, 32, 2)
+        lazy_module = nn.LazyConv1d(32, 2)
+        with self.assertRaisesRegex(RuntimeError, 'shape of an uninitialized'):
+            module.load_state_dict(lazy_module.state_dict())
+
+    @suppress_warnings
+    def test_lazy_conv2d(self):
```

Comment:
Since you run the same tests for each conv type, would it be possible to refactor this into a test function that takes the conv module as input so that you can re-use it?

Reply:
Resolved test code duplication in 7a2cc32!